### DISCUSSION TITLE
Providing a condensed "engineering" report format

### DISCUSSION THREAD

**Discussion by RichardPfr** [pyroll-project/pyroll-core#217] [Q&A]
> [!NOTE]
> This post is supposed to be a collection of ideas and a place to discuss them, whether these ideas are feasible or not. For the actual development of this see pyroll-project/pyroll-report#19

### Idea
As was hinted at in recent discussions a more condensed report format may be necessary which, in contrast to the current report, only shows the most substantial dimensions. As an "**_engineering_**" report it should provide a quick overview over the most needed dimensions, optimally also in graphically form. If more informations are needed one can still decide to output the current "_**scientific**_" format which contains all dimensions, as is the case currently.

One limitation of this might necessary be that this "_engineering_" report is less flexible and more static. The dimensions to be shown may need to be set beforehand and therefore may not be able to be changed easily. While the "_scientific_" one extends dynamically with additional packages beeing loaded, an "_**engineering**_" one like this might not.
This is all to be seen when developing it though.

### Question
For this reason we ask you, what...
- ... features you want to see
- ... ideas you have 
- ... and contents you think such an "_engineering_" report definitely should have

**Below we already gathered some ideas from our side.** 
**Please feel free to also give us your opinion on any ideas we or others posted.**

**Comment by RichardPfr** [comment]
Here some features, ideas and contents to be included in such a report from our/ my side:
### possible features
- show dimensions in engineering units (see pyroll-project/pyroll-report#8 and pyroll-project/pyroll-core#35)
- output graphs for all chosen dimensions (below)
- option to reformat the report to be optimised for printing on A4
- make the web interface of the report more interactive
  - pass/ groove figures as frames in an animation like [here](https://plotly.com/python/animations/)

### contents
- roll forces 
- roll-/ motor torque (after gear reductions)
- motor power usage (with power usage diagrams?)
- elongation

**Reply by sag-kalibreur** [reply]
### Here are some features that I consider to be important information:

- engineering units (input `and` output)
- strain rates
- strain
- temperature evolution (core, mean, surface)
- roll tourque / motor tourque
- roll load
- motor power
- reductions
- bite angles
- spread coefficients
- elongation / elongation efficiency
- filling ratios
- roll diameter
- working diameter
- velocity / rotational frequency
- heights / widths / cross sections / roll gaps
- plots of incoming and outgoing profil

This data can be used to evaluate a pass sequence in detail. Another good feature would be a conversion to PDF format (A4).
The data can be in tabular form and/or as information in the corresponding plots.

Please share your opinion on this...
